src/NUnitEngine/nunit.engine/Services/TestAgency.cs

```diff
@@ -154,10 +155,10 @@ private Guid LaunchAgentProcess(TestPackage package)
             // Set options that need to be in effect before the package
             // is loaded by using the command line.
             string agentArgs = "--pid=" + Process.GetCurrentProcess().Id.ToString();
+            if (traceLevel != "Off")                    //Pass --trace as first non-optional arg as it intialises log
+                agentArgs += " --trace:" + traceLevel;
```

Comment:
This order-dependency seems fragile. See my comment in Program.cs.

Reply:
Yeah - I protected against a crash by checking the log was initialised - but this was mainly me being lazy and making the minimal required changes. I can do this bit better.